idseq_dag/util/log.py

```diff
@@ -34,6 +39,70 @@ def write(message, warning=False, flush=True):
         if flush:
             sys.stdout.flush()
 
+def log_event(event_name, values=None, start_time=None, warning=False, flush=True):
+    '''
+    Write a new log event.
+
+    Parameters:
+    event_name (str): name of the event
+    values(dict): Optional. Values associated to that event. Will be logged in json format.
+    start_time(datetime): Optional. If given, it will calc the elapsed time from this datetime to now and write it to the log line.
+
+    Returns:
+    datetime: Now. It can be used to pass to parameter start_time in a future log_event call.
+
+    Example:
+    start = log_event("downloaded_started", {"file":"abc"})
+    # ... download your file here ...
+    log_event("downloaded_completed", {"file":"abc"}, start_time=start)
+    '''
+    fmt_values = "{}" if values is None else json.dumps(values)
+    if start_time is None:
+        fmt_message = "Event '%s' %s" % (event_name, fmt_values)
+    else:
+        duration = (datetime.datetime.now()-start_time).total_seconds()
+        fmt_message = "Event '%s' %s (%.1f seconds)" % (event_name, fmt_values, duration)
```

Comment:
We were considering making all log messages in json format for easier parsing downstream.

Reply:
About JSON: this is super easy to change, but I haven't completely done that yet because there are a lot of "non-json" logs floating around. How does Datadog handle log files that mix JSON and text?

Reply:
Not sure. You might want to sync with @jshoe when he comes back.
I think Datadog will still index the free text, but will recognize everything as the same field. 
You can either start changing to JSON or wait to sync with Jonathan and understand Datadog better to do that. I would go with the first option but is up to you.

Reply:
I feel like going to full JSON might be part of a separate PR. We probably need to define some additional standards about what fields will be part of the json in which format, name, (ex: timestamp).